src/util/metadata/mark_bit.rs

```diff
@@ -0,0 +1,16 @@
+use crate::util::ObjectReference;
+use crate::vm::VMBinding;
+use crate::vm::VMLocalMarkBitSpec;
+use std::sync::atomic::Ordering;
+
+impl VMLocalMarkBitSpec {
+    /// Set the mark bit for the object to 1
+    pub fn set<VM: VMBinding>(&self, object: ObjectReference, ordering: Ordering) {
```

Comment:
it is probably better to rename the methods to `mark`/`is_marked`.

Reply:
Sounds good to me. I was avoiding using `mark`, as we may use cyclic mark in the future. In that case, setting to `1` does not mean marking. But we could leave that to future refactoring.